src/main/java/com/thinkful/noteful/auth/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,71 @@
+package com.thinkful.noteful.auth;
+
+import static com.thinkful.noteful.auth.SecurityConstants.EXPIRATION_TIME;
+import static com.thinkful.noteful.auth.SecurityConstants.HEADER_NAME;
+import static com.thinkful.noteful.auth.SecurityConstants.JWT_SECRET;
+import static com.thinkful.noteful.auth.SecurityConstants.TOKEN_PREFIX;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.thinkful.noteful.users.User;
+
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.springframework.beans.factory.annotation.Autowired;
```

Comment:
`import org.springframework.beans.factory.annotation.Autowired;` is no longer needed - smoke it.